docs/designers-developers/developers/block-api/block-edit-save.md

```diff
@@ -226,6 +226,13 @@ For most blocks, the return value of `save` should be an [instance of WordPress
 
 _Note:_ While it is possible to return a string value from `save`, it _will be escaped_. If the string includes HTML markup, the markup will be shown on the front of the site verbatim, not as the equivalent HTML node content. If you must return raw HTML from `save`, use `wp.element.RawHTML`. As the name implies, this is prone to [cross-site scripting](https://en.wikipedia.org/wiki/Cross-site_scripting) and therefore is discouraged in favor of a WordPress Element hierarchy whenever possible.
 
+_Note:_ The save function should be a pure function that depends only on the attributes used to invoke it.
+It can not have any side effect or retrieve information from another source, e.g. it is not possible to use the data module inside it `wp.data.select...`.
```

Comment:
I think the mention of "`wp.data.select...`" could help clarify what's referred to, but for what it's worth, we should discourage this use of the window global in blocks _anywhere_, since it cannot account for separate data registries, which will become necessary for e.g. embedded editors (#7119), multiple editors on a page.